api/views/admin_auth.py

```diff
@@ -62,6 +81,118 @@ def register():
         return make_response(jsonify(response_object), 400)
 
 
+@admin_auth.route('/confirmation/<confirmation_id>', methods=['GET'])
+def confirm_registration(confirmation_id):
+    """ Endpoint that verifies user's signup process through a confirmation ID from email """
+
+    def is_valid_uuid(val):
+        try:
+            uuid.UUID(str(val))
+            return True
+        except ValueError:
+            return False
+
+    if not is_valid_uuid(confirmation_id):
+        response_object = {
+            "status": False,
+            "message": "Invalid confirmation parameter."
+        }
+
+        return make_response(jsonify(response_object), 401)
+
+    user = users.update_one(
+        {"emailConfirmationId": confirmation_id},
+        {
+            "$set": {"emailConfirmed": True},
+            "$unset": {"emailConfirmationId": ""}
+        },
+        upsert=False
+    )
+
+    if not user.acknowledged:
+        response_object = {
+            "status": False,
+            "message": "Invalid ID. Please register a new user or resend the verification email."
+        }
+        return make_response(jsonify(response_object), 401)
+
+    if user.modified_count == 0:
+        response_object = {
+            "status": False,
+            "message": "User already has been registered. Please register a new user or resend the verification email."
+        }
+        return make_response(jsonify(response_object), 401)
+
+    response_object = {
+        "status": True,
+        "message": "Email verified."
+    }
+    return "verified"
+    return response_object(jsonify(response_object), 200)
+
+
+@admin_auth.route('/resend-confirmation', methods=['POST'])
+def resend_confirmation():
+    """ Endpoint that resends confirmation email to the user with a new ID """
+    data = validate_email(request.get_json())
+
+    if data['ok']:
+        data = data['data']
+        email = data["email"]
+
+        user = users.find_one({
+            "email": email
+        })
+
+        if not user:
+            response_object = {
+                "status": False,
+                "message": "Email not found. Please register a new account."
+            }
+            return make_response(jsonify(response_object), 400)
+
+        if user["emailConfirmed"]:
+            response_object = {
+                "status": False,
+                "message": "User has already been verified."
+            }
+            return make_response(jsonify(response_object), 400)
+
+        # Generates a new UUID
+        new_confirmation_id = str(uuid.uuid4())
+
+        # Updated the new UUID in the DB
+        user = users.update(
+            {"email": email},
+            {
+                "$set": {"emailConfirmationId": new_confirmation_id},
+            },
+            upsert=False
+        )
+
```

Comment:
put the update after re-sending the confirmation email?

Reply:
So if you see the email sending middleware, the uuid is critical to it meaning that data should be updated on the DB first which then, a user should be able to confirm their account. Imagine if an email gets sent out first, the user confirms their email before the DB gets updated - let me know if you need more explanation with this workflow!

Reply:
ye make sense!